src/types/validate_test.go

```diff
@@ -0,0 +1,506 @@
+			action: ZarfComponentAction{},
+		},
+		{
+			name: "cmd and wait both set, nothing in wait",
+			action: ZarfComponentAction{
+				Cmd:  "ls",
+				Wait: &ZarfComponentActionWait{},
+			},
+			expectedErrs: []string{
+				fmt.Sprintf(lang.PkgValidateErrActionCmdWait, "ls"),
+				lang.PkgValidateErrActionClusterNetwork,
+			},
+		},
+		{
+			name: "cluster and network both set",
+			action: ZarfComponentAction{
+				Wait: &ZarfComponentActionWait{Cluster: &ZarfComponentActionWaitCluster{}, Network: &ZarfComponentActionWaitNetwork{}},
+			},
+			expectedErrs: []string{fmt.Sprintf(lang.PkgValidateErrActionClusterNetwork)},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.action.Validate()
+			if tt.expectedErrs == nil {
+				require.NoError(t, err)
+				return
+			}
+			errs := strings.Split(err.Error(), "\n")
+			require.ElementsMatch(t, tt.expectedErrs, errs)
+		})
+	}
+}
+
+func TestValidateZarfComponent(t *testing.T) {
+	t.Parallel()
+	absPath, err := filepath.Abs("abs")
+	require.NoError(t, err)
+	tests := []struct {
+		component    ZarfComponent
+		expectedErrs []string
+		name         string
+	}{
+		{
+			name: "valid path",
+			component: ZarfComponent{
+				Name: "component1",
+				Import: ZarfComponentImport{
+					Path: "relative/path",
+				},
+			},
+			expectedErrs: nil,
+		},
+		{
+			name: "valid URL",
+			component: ZarfComponent{
+				Name: "component2",
+				Import: ZarfComponentImport{
+					URL: "oci://example.com/package:v0.0.1",
+				},
+			},
+			expectedErrs: nil,
+		},
+		{
+			name: "neither path nor URL provided",
+			component: ZarfComponent{
+				Name: "neither",
+			},
+			expectedErrs: []string{
+				fmt.Sprintf(lang.PkgValidateErrImportDefinition, "neither", "neither a path nor a URL was provided"),
+			},
+		},
+		{
+			name: "both path and URL provided",
+			component: ZarfComponent{
+				Name: "both",
+				Import: ZarfComponentImport{
+					Path: "relative/path",
+					URL:  "https://example.com",
+				},
+			},
+			expectedErrs: []string{
+				fmt.Sprintf(lang.PkgValidateErrImportDefinition, "both", "both a path and a URL were provided"),
+			},
+		},
+		{
+			name: "absolute path provided",
+			component: ZarfComponent{
+				Name: "abs-path",
+				Import: ZarfComponentImport{
+					Path: absPath,
+				},
+			},
+			expectedErrs: []string{
+				fmt.Sprintf(lang.PkgValidateErrImportDefinition, "abs-path", "path cannot be an absolute path"),
+			},
+		},
+		{
+			name: "invalid URL provided",
+			component: ZarfComponent{
+				Name: "bad-url",
+				Import: ZarfComponentImport{
+					URL: "https://example.com",
+				},
+			},
+			expectedErrs: []string{
+				fmt.Sprintf(lang.PkgValidateErrImportDefinition, "bad-url", "URL is not a valid OCI URL"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.component.Validate()
+			if tt.expectedErrs == nil {
```

Comment:
I nitted Wayne about it may be better to make expectedErrs a bool, but it's up to you.

Reply:
I think this is better to avoid situations where expectedErrs is not empty but the bool is false.